currencycom/asyncio/websockets.py

```diff
@@ -0,0 +1,241 @@
+import asyncio
+import json
+import logging
+import time
+import websockets
+
+from random import random
+from datetime import datetime
+from typing import Optional
+
+from ..client import CurrencycomClient
+
+
+class ReconnectingWebsocket:
+    MAX_RECONNECTS = 5
+    MAX_RECONNECT_SECONDS = 60
+    MIN_RECONNECT_WAIT = 0.5
+    TIMEOUT = 10
+    PING_TIMEOUT = 5
+
+    def __init__(self, loop, client, coro):
+        self._loop = loop
+        self._log = logging.getLogger(__name__)
+        self._coro = coro
+        self._reconnect_attempts = 0
+        self._conn = None
+        self._connect_id = None
+        self._socket = None
+        self._request = {
```

Comment:
Let's move this var to `send_message` as it shouldn't be stored across the object